presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -831,6 +939,14 @@ public static boolean isTypeOnlyCoercion(Type actualType, Type expectedType)
                 return Optional.of(new MapType(keyType.get(), valueType.get()));
             }
         }
+        if (firstType instanceof DecimalType && secondType instanceof DecimalType) {
+            DecimalType firstDecimal = (DecimalType) firstType;
+            DecimalType secondDecimal = (DecimalType) secondType;
+            int targetScale = Math.max(firstDecimal.getScale(), secondDecimal.getScale());
+            int targetPrecision = Math.max(firstDecimal.getPrecision() - firstDecimal.getScale(), secondDecimal.getPrecision() - secondDecimal.getScale()) + targetScale;
```

Comment:
extract firstDecimal.getPrecision() - firstDecimal.getScale as firstDecimalDigitsBeforeDot. the same for second decimal